restorer.go

```diff
@@ -35,13 +38,25 @@ func (r *Restorer) Restore(cache Cache) error {
 
 	var g errgroup.Group
 	for _, buildpack := range r.Buildpacks {
+		cachedLayers := meta.MetadataForBuildpack(buildpack.ID).Layers
+
+		var cachedFn func(bpLayer) bool
+		if api.MustParse(buildpack.API).Compare(api.MustParse("0.6")) >= 0 {
```

Comment:
What about changing `forCached` into `forCached(bpAPI, cachedLayers) `? Or maybe a new function that is restorer specific? The `forCached` isn't located here and it isn't immediately clear to me why we branch the logic only in restorer and not in analyzer.

Reply:
I thought about that, but `findLayers` and `forCached` are also called by the exporter and it felt weird to change the signature when the exporter doesn't care about the bp api, only the restorer (the exporter "trusts" that the types information - or lack thereof - in each `<layer>.toml` is what the buildpack intended; since the restorer is running before the build phase, it can't do that).